### ISSUE TITLE
Create an inventory system

### ISSUE THREAD

**Issue by blalasaadri** [CommandLineHeroes/hero-engine#24]
Most adventure games have an inventory of some kind. Detailed descriptions of the various options can be found in [The Transfer and Extension of Concepts from Classical Point & Click Adventures for Smartphones and Tablets, chapter 2.4.3 and Appendix A.4](https://drive.google.com/file/d/0B8CqGc-YuVkVdnZQZE5PTUJJU28/view), but they basically boil down to:

- only one item can be held at a time
- a large (sometimes unlimited) number of items can be held at any time

The second option has variations in visibility; in some cases the inventory items are always visible, in other cases they are only visible upon request.

Here is what that may look like:
- The inventory system in "Simon the Sorcerer" (here in the iOS version, though the interface is identical to any other version) is always visible and could hold an unlimited number of items. ![The inventory system in "Simon the Sorcerer" (iOS version)](https://drive.google.com/uc?id=0B8CqGc-YuVkVZ1ViYXRNQ1lpR2M)
- The inventory system in "Edna & Harvey: The Breakout" (PC version) is only visible upon request. ![The inventory system in "Edna & Harvey: The Breakout" for the PC](https://drive.google.com/uc?id=0B8CqGc-YuVkVcDJjczE4ZjU0bE0) In the iOS version, this was implemented slightly differently but the essential functionality stayed the same.
- In "Puzzle Quest" there is no inventory as such; instead, there is a notebook containing topics about which the protagonist can try to speak with NPCs. ![The inventory system in "Puzzle Quest".](https://drive.google.com/uc?id=0B8CqGc-YuVkVWWZKckFBWXc5dFk)
- In "Plum" (a game prototype for a [contest](https://www.visionaire-studio.net/forum/thread/stepping-across-the-line-adventure-game-contest-20-08-21-09-2012-/1/) held by the people behind Visionaire Studio in 2012), the main character could only ever carry a single item above his head. (Windows only, and since I don't currently have a Windows PC available... If interested, I've uploaded the game [here](https://drive.google.com/file/d/1420EH7UFafoRC7WntSECnM1fYTAgiTGv/view?usp=sharing).)

Nowadays, most modern adventure games feature an inventory which can hold several items (as many as are required to solve the tasks at hand) which can be displayed on request but is normally hidden behind an icon of some kind. This would be my suggestion for what the _Hero Engine_ should support.

As soon as a decision has been made, we'll need a reference room implementation of such a system.

**Comment by Qazwsxedcr** [comment]
Whatever is chose, it should definitely tie into the game. Like for "Puzzle Quest" it was a great idea like holding up a notepad in real life because you need it to write important info down. Since the game uses an integreated command line, maybe it will be like "Simon the Sorcerer"? where the terminal is always visible and we hit certain commands to show the inventory and other stuff?

**Comment by gigglearrows** [comment]
I agree with @steaksauce-. The functionality for storing items, and maby add parameters for maximum amount of items that can be stored in the inventory. And ofc add functionality to fetch a list of items in inventory, remove items etc. Then the game developer can choose how to display the inventory and how the items shall be used/consumed
